exec/java-exec/src/main/java/org/apache/drill/exec/store/sys/store/provider/LocalPersistentStoreProvider.java

```diff
@@ -57,14 +59,14 @@ public LocalPersistentStoreProvider(final DrillConfig config) throws StoreExcept
   }
 
   @Override
-  public <V> PersistentStore<V> getOrCreateStore(PersistentStoreConfig<V> storeConfig) {
+  public <V> PersistentStore<V> getOrCreateStore(PersistentStoreConfig<V> storeConfig) throws StoreException {
     switch(storeConfig.getMode()){
     case BLOB_PERSISTENT:
     case PERSISTENT:
-      if (enableWrite) {
+      if (!useEphemeral) {
```

Comment:
This is changing the default behavior, as enableWrite/useEphemeral is true by default.